control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -770,23 +774,36 @@ func (r *reconciler) reconcileIngressController(ctx context.Context, hcp *hyperv
 			errs = append(errs, fmt.Errorf("failed to retrieve guest cluster ingress NodePort: %w", err))
 		}
 
-		hcpNamespace := hcp.Namespace
+		var targetClient client.Client
+		var namespace string
+		// If external infra cluster is being used - use infra client and external cluster namespace
+		// rather than mgmt client and hosted control plane namespace
+		if hcp.Spec.Platform.Kubevirt.ExtClusterKubeConfigSecret != nil &&
+			hcp.Spec.Platform.Kubevirt.ExtClusterNamespace != "" {
+			r.externalInfra = true
+			r.externalInfraNamespace = hcp.Spec.Platform.Kubevirt.ExtClusterNamespace
```

Comment:
the problem with that is the fact that the source of `externalInfraNamespace` is in the HCP object, and when HCCO is initialized the HCP object is not yet visible to it.
therefore i set that value when it was needed and the hcp is available to consume.

regarding always use `r.infraClient` - yes, but we will still have the branching because of the target namespace detection.

Reply:
i left a more detailed follow for this on another comment. We can probably avoid the `r.externalInfra` and `r.externalInfranamespace` vars by inspecting the hcp object during creation and deletion

Reply:
done.